packages/edit-post/src/store/selectors.js

```diff
@@ -583,12 +583,27 @@ export const getEditedPostTemplate = createRegistrySelector(
 		}
 
 		const post = select( editorStore ).getCurrentPost();
-		if ( post.link ) {
-			return select( coreStore ).__experimentalGetTemplateForLink(
-				post.link
-			);
+		let slugToCheck;
```

Comment:
I think this function is actually a duplicate of a similar behavior `useResolveEditedEntityAndContext` in the edit-site package. So I think we should probably apply the same fix here which might fix an issue @ellatrix experienced with the wrong template being loaded in the site editor for draft pages.

Reply:
>So I think we should probably apply the same fix here

You're talking about the slug to check logic and you  meant `there` instead of `here` I reckon, right?

Reply:
yes, I mean there :)

Reply:
Done